src/SingleProject/Resizetizer/src/GenerateSplashAndroidResources.cs

```diff
@@ -18,53 +24,96 @@ public class GenerateSplashAndroidResources : Task, ILogger
 		[Required]
 		public ITaskItem[] MauiSplashScreen { get; set; }
 
+		public string InputsFile { get; set; }
+
 		public override bool Execute()
 		{
-			var splash = MauiSplashScreen[0];
+			Log.LogMessage(MessageImportance.Low, $"Splash Screen: Intermediate Path " + IntermediateOutputPath);
+
+			var splash = MauiSplashScreen?.FirstOrDefault();
+			if (splash is null)
+			{
+				Log.LogMessage(MessageImportance.Low, $"Splash Screen: No images found.");
+				return true;
+			}
 
-			var info = ResizeImageInfo.Parse(splash);
+			try
+			{
+				var info = ResizeImageInfo.Parse(splash);
 
-			var tools = SkiaSharpTools.Create(info.IsVector, info.Filename, info.BaseSize, info.Color, info.TintColor, this);
+				var resizer = new Resizer(info, IntermediateOutputPath, this);
 
-			WriteColors(info, tools);
-			WriteDrawable(info, tools);
-			WriteDrawable_v31(info, tools);
+				WriteImages(resizer);
+				WriteColors(resizer);
+				WriteDrawable(resizer);
+				WriteDrawable_v31(resizer);
 
-			return !Log.HasLoggedErrors;
+				return !Log.HasLoggedErrors;
+			}
+			catch (Exception ex)
+			{
+				Log.LogError(null, "MAUI0000", null, null, 0, 0, 0, 0, ex.ToString());
+				return false;
+			}
 		}
 
 		static readonly XmlWriterSettings Settings = new XmlWriterSettings { Indent = true };
 		const string Namespace = "http://schemas.android.com/apk/res/android";
 		const string Comment = "This file was auto-generated by .NET MAUI.";
 		const float PreferredImageSize = 108f;
 
-		void WriteColors(ResizeImageInfo splash, SkiaSharpTools tools)
+		private void WriteImages(Resizer resizer)
+		{
+			if (resizer.Info.Resize)
+			{
+				foreach (var dpi in DpiPath.Android.Image)
+				{
+					Log.LogMessage(MessageImportance.Low, $"Splash Screen Resize: " + dpi);
+					resizer.Resize(dpi, InputsFile);
+				}
+			}
+			else
+			{
+				var dpi = DpiPath.Android.Original;
+
+				Log.LogMessage(MessageImportance.Low, $"Splash Screen Copy: " + dpi);
+				resizer.CopyFile(dpi, InputsFile);
+			}
+		}
+
+		void WriteColors(Resizer resizer)
 		{
 			var dir = Path.Combine(IntermediateOutputPath, "values");
 			Directory.CreateDirectory(dir);
 
 			var colorsFile = Path.Combine(dir, "maui_colors.xml");
+
+			Log.LogMessage(MessageImportance.Low, $"Splash Screen Colors: " + colorsFile);
+
 			using var writer = XmlWriter.Create(colorsFile, Settings);
 			writer.WriteComment(Comment);
 			writer.WriteStartElement("resources");
 
-			if (splash.Color is not null)
+			if (resizer.Info.Color is not null)
```

Comment:
Can `.Info` be null? Does it need to be `resizer.Info?.Color`?

Reply:
Info cannot be null as it is validated and passed to the resizer constructor.